### PR TITLE
Move secrets to status reconciler

### PR DESCRIPTION
**Pull request by bcrochet**
Since we are just checking that the secrets exist and are in a correct
key, this is just a status update.

Signed-off-by: Brad P. Crochet <brad@redhat.com>

### CODE REVIEW

File: internal/reconcilers/status.go [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+		r.conditionStatus(true),
+		"AsExpected",
+		fmt.Sprintf("%s with name %s found", indexType, indexName),
+		readyCondition))
+
+	return false, nil
+}
+
+func (r *StatusReconciler) reconcileSecretStatus(ctx context.Context, pipeline *certv1alpha1.OperatorPipeline, secretType, secretName, secretKey string) (bool, error) {
+	// Kubeconfig Secret
```
Comment (acornett21) [inline internal/reconcilers/status.go:175]: Am I missing something? This isn't just for the kubeconfig right?

File: internal/reconcilers/status.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+	readyCondition := v1.Condition{
+		Type:               fmt.Sprintf("%sReady", secretType),
+		ObservedGeneration: pipeline.Generation,
+		Status:             v1.ConditionUnknown,
+	}
+	log := r.Log.WithValues("status.observedGeneration", pipeline.Generation)
+	secret := &corev1.Secret{}
+	err := r.Client.Get(ctx, types.NamespacedName{Namespace: pipeline.Namespace, Name: secretName}, secret)
+	if err != nil && !apierrors.IsNotFound(err) {
+		log.WithValues("kubeconfigsecret", types.NamespacedName{Namespace: pipeline.Namespace, Name: secretName}).
```
Comment (acornett21) [inline internal/reconcilers/status.go:185]: Shouldn't this be `secreteType` or `secreteName` here?

File: internal/reconcilers/status.go (hunk 1 above)
Comment (acornett21) [inline internal/reconcilers/status.go:175]: Am I missing something? This isn't just for the kubeconfig right?

File: internal/reconcilers/status.go [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+			return requeue, err
+		}
+	}
+
+	if len(pipeline.Spec.GithubSSHSecretName) > 0 {
+		requeue, err = r.reconcileSecretStatus(ctx, pipeline, "GithubSSHSecret", pipeline.Spec.GithubSSHSecretName, defaultGithubSSHSecretKeyName)
+		if requeue || err != nil {
+			return requeue, err
+		}
+	}
```
Comment (acornett21) [inline internal/reconcilers/status.go:88]: Non Blocking: I know this is the order we were checking before, but maybe it makes sense to move this to after line 99, so this file is easier to read top down.

File: internal/reconcilers/status.go (hunk 2 above)
Comment (bcrochet) [reply internal/reconcilers/status.go:185]: Good catch.

File: internal/reconcilers/status.go (hunk 1 above)
Comment (bcrochet) [reply internal/reconcilers/status.go:175]: It is not. Missed that on refactor.

File: internal/reconcilers/status.go (hunk 3 above)
Comment (bcrochet) [reply internal/reconcilers/status.go:88]: I actually moved it to L71, to put it immediately after the other Github secret.

**Review by acornett21 — APPROVED** [review]

**Review by jomkz — APPROVED** [review]
/lgtm

**Pull request merged**